scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -371,30 +367,21 @@ trait SequenceFileScheme extends SchemedSource {
 }
 
 /**
- * Ensures that a _SUCCESS file is present in every directory included by a glob,
- * as well as the requirements of [[FileSource.pathIsGood]]. The set of directories to check for
- * _SUCCESS
- * is determined by examining the list of all paths returned by globPaths and adding parent
- * directories of the non-hidden files encountered.
- * pathIsGood should still be considered just a best-effort test. As an illustration the following
- * layout with an in-flight job is accepted for the glob dir*&#47;*:
- * <pre>
- *   dir1/_temporary
- *   dir2/file1
- *   dir2/_SUCCESS
- * </pre>
+ * Uses _SUCCESS files instead of the presence of non-hidden files to determine if a path is good.
  *
- * Similarly if dir1 is physically empty pathIsGood is still true for dir*&#47;* above
+ * Requires that:
+ * 1) Every matched, non-hidden directory contains a _SUCCESS file
+ * 2) Every matched, non-hidden file's parent directory contain a _SUCCESS file
  *
- * On the other hand it will reject an empty output directory of a finished job:
- * <pre>
- *   dir1/_SUCCESS
- * </pre>
+ * pathIsGood should still be considered just a best-effort test. There are still cases where this is
+ * not a sufficient test for correctness. See <ticket>
```

Comment:
well, if you supply the glob: `a/b/c/{x,y,z}` it is clear to a human that you expect x, y, and z to all exist and have SUCCESS files. But actually, what happens is:

1) If only one of {x,y,z} exists, and it has a success file, we will call this "good" because the non-existent dirs won't return any files and so we won't validate them
2) if some of the dirs do exist, but have no success files, then it will be "not good"

The only way around this is to try and parse the glob syntax looking for things like {x,y,z}, or instead separating these paths out to 3 separate globs in `hdfsPaths` instead of consoldiating them to 1 glob (which is what TimePathedSource does)

I need to file <ticket> w/ the details and reference it here.